<launch>
	<!-- RTAB-MAP SENSORY-MOTOR VERSION -->
	<!-- WARNING : Database is automatically deleted on each startup -->
	<!--           See "delete_db_on_start" option below... -->

	<!-- image_to_sm parameters -->
	<param name="resize_image_width" value="640" type="int"/>
	<param name="resize_image_height" value="480" type="int"/>
	
	<!-- Nodes -->	
	<!-- cmd_vel_a has priority on cmd_vel_b -->
	<node name="abtr_velocity" pkg="rtabmap" type="abtr_velocity_node">
		<remap from="cmd_vel_a" to="user/cmd_vel" />
		<remap from="cmd_vel_b" to="rtabmap/cmd_vel" />
		<remap from="cmd_vel" to="cmd_vel" />
	</node>
	
	<node name="rtabmap_output" pkg="rtabmap" type="output_node">
		<remap from="rtabmap/cmd_vel" to="rtabmap/cmd_vel" />
		<remap from="rtabmap_info" to="rtabmap_info" />
		<remap from="rtabmap_info_x" to="rtabmap_info_x" />
	</node>
		
	<node name="rtabmap_core" pkg="rtabmap" type="core_node" output="screen" args="--delete_db_on_start">
		<remap from="sm_state" to="sm_state" />
		<remap from="rtabmap_info" to="rtabmap_info" />
		<remap from="rtabmap_info_x" to="rtabmap_info_x" />
	</node>

	<node name="rtabmap_input" pkg="rtabmap" type="input_node">
		<remap from="cmd_vel" to="cmd_vel" />
		<remap from="sm_state" to="sm_state" />
		<remap from="camera_data" to="camera_data" />
	</node>
	
	<node name="image_to_sms" pkg="rtabmap" type="image_to_sms_node">
		<remap from="image" to="image" />
		<remap from="sm_state" to="camera_data" />
	</node>
</launch>